### PR TITLE
Fixing un-used imports sonar violation.

### PR DESCRIPTION
**Pull request by krmahadevan**
Fixes # .

### Did you remember to?

- [ ] Add test case(s)
- [ ] Update `CHANGES.txt`

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @cbeust / @juherr - Can you please help get this merged ?

**Comment by juherr** [comment]
@krmahadevan could you check the changes with SonarLint (configured with the sonarqube.com instance)? 
I think sonar will still complain against some changes ;)

**Comment by krmahadevan** [comment]
@juherr can you please help guide me how do I that ? ( Pls bear with me if that sounds silly )

**Comment by juherr** [comment]
@krmahadevan No prob!
1. Install SonarLint: http://www.sonarlint.org/ (it will highlight some issues)
2. Attach it to the external configuration (Connected Mode with http://sonarqube.com as  server)

Step 2 may fail if you have not an account, try to create one in that case.

**Comment by juherr** [comment]
@krmahadevan I've just provided you some rights (administer issues) on testng project on sq.com

**Comment by cbeust** [comment]
@krmahadevan Let me know when you think this is ready to be merged.

**Comment by krmahadevan** [comment]
@cbeust - I think this is ready for review and merge.

@juherr - Thanks for sharing that information on the sonarlint. I have tried to fix some of the recommendations. I guess this will require multiple iterations so that its easy to focus on the changeset for me and then fix anything if the tests break. So I haven't run through all of the recommendations.

File: src/main/java/org/testng/internal/ClassHelper.java [hunk 1]
```diff
@@ -172,26 +174,32 @@ private static void logClassNotFoundError(String className, Exception ex) {
...
+          Utils.log(CLASS_HELPER, 2, "JUnit 4 was not found on the classpath");
           try {
               //fallback to JUnit 3
               Class.forName("junit.framework.Test");
-              IJUnitTestRunner tr = (IJUnitTestRunner) ClassHelper.forName(JUNIT_TESTRUNNER).newInstance();
-              tr.setTestResultNotifier(runner);
-
-              return tr;
+              Class<?> clazz =ClassHelper.forName(JUNIT_TESTRUNNER);
+              if (clazz != null) {
```
Comment (juherr) [inline src/main/java/org/testng/internal/ClassHelper.java]: If clazz is null, I think we should have the testng exception (no sure it was the previous behavior)

**Review by juherr — CHANGES_REQUESTED** [review]
Wow! You went a bit deeper than what I expected :)

Be careful because every rules may be debatable and changes may become huge quickly.

In fact, my idea is not fixing everything but, at least, having no issue on new code (aka leak period on https://sonarqube.com/dashboard/index?id=org.testng%3Atestng)
SonarQube should be able to comment pull request automaticaly soon (I hope so!)

BTW, this PR looks good except 2 changes which are potential regressions.

**Comment by cbeust** [comment]
@krmahadevan Can you please address @juherr's two comments? I'll merge after that.

Thanks!

File: src/main/java/org/testng/ClassMethodMap.java [hunk 2]
```diff
@@ -17,26 +17,26 @@
  * @author <a href='mailto:the[dot]mindstorm[at]gmail[dot]com'>Alex Popescu</a>
  */
 public class ClassMethodMap {
-  private Map<Object, List<ITestNGMethod>> m_classMap = Maps.newHashMap();
+  private Map<Object, List<ITestNGMethod>> classMap = Maps.newHashMap();
   // These two variables are used throughout the workers to keep track
   // of what beforeClass/afterClass methods have been invoked
-  private Map<ITestClass, Set<Object>> m_beforeClassMethods = Maps.newHashMap();
-  private Map<ITestClass, Set<Object>> m_afterClassMethods = Maps.newHashMap();
+  private Map<ITestClass, Set<Object>> beforeClassMethods = Maps.newHashMap();
```
Comment (cbeust) [inline src/main/java/org/testng/ClassMethodMap.java]: By the way, why that renaming of the field? This should be a separate PR (assuming we want to do that at all).

Comment (juherr) [inline src/main/java/org/testng/ClassMethodMap.java]: `m_` is a very old naming convention. Maybe we can consider using a new convention on a new or modified code?
I think @krmahadevan was in a good mood and it can be difficult to ignore all the highlighted code from sonarlint once you start to fix it :p
BTW, do you think it should be reverted?

**Comment by juherr** [comment]
@krmahadevan It looks the build on openjdk7 is always failing with this pull request. I think it is a side effect but could you try to have a look?

**Comment by juherr** [comment]
@krmahadevan Forget it, it passed this time!

**Comment by juherr** [comment]
@cbeust the 2 potential issues were not yet fixed ;)

**Comment by cbeust** [comment]
Oops sorry I misread. I'll let @krmahadevan fix them in a separate PR.

File: src/main/java/org/testng/ClassMethodMap.java (hunk 2 above)
Comment (krmahadevan) [inline src/main/java/org/testng/ClassMethodMap.java]: @juherr Lol.. Yeah you were right on that... I got a bit carried away and obsessed with driving down the sonar cube count down... I must admit, its a bit difficult to keep sonar cube happy :)

**Comment by krmahadevan** [comment]
@juherr and @cbeust - I will raise a new PR for the potential issues fixes that were called out here.

**Comment by krmahadevan** [comment]
@cbeust and @juherr - lets say, I spent some time in renaming all variables that start with `m_` and removing the prefix. Do you think it can get merged ? I wanted to confirm with you folks before I do that, because it will be a bit of an effort :)
